code/game/objects/items/stacks/stack.dm

```diff
@@ -201,31 +201,31 @@
 			to_chat(usr, "<span class='warning'>\The [R.title] won't fit here!</span>")
 			return FALSE
 
-		if(R.one_per_turf && (locate(R.result_type) in usr.drop_location()))
+		if(R.one_per_turf && (locate(R.result_type) in get_turf(src)))
 			to_chat(usr, "<span class='warning'>There is another [R.title] here!</span>")
 			return FALSE
 
-		if(R.on_floor && !istype(usr.drop_location(), /turf/simulated))
+		if(R.on_floor && !istype(get_turf(src), /turf/simulated))
 			to_chat(usr, "<span class='warning'>\The [R.title] must be constructed on the floor!</span>")
 			return FALSE
 
-		if(R.no_cult_structure && (locate(/obj/structure/cult) in usr.drop_location()))
+		if(R.no_cult_structure && (locate(/obj/structure/cult) in get_turf(src)))
 			to_chat(usr, "<span class='warning'>There is a structure here!</span>")
 			return FALSE
 
 		if(R.time)
 			to_chat(usr, "<span class='notice'>Building [R.title] ...</span>")
-			if(!do_after(usr, R.time, target = usr))
+			if(!do_after(usr, R.time, target = src.loc))
```

Comment:
Normally it's preferred not to use `src.` anywhere since it's always implied, but in this case I think it makes the code more readable, so sure.